diff --git a/report.c b/report.c
index 95f5872..c787b28 100644
--- a/report.c
+++ b/report.c
@@ -25,10 +25,10 @@ static const char HEADER[] =
 static void set_text(char *dst, size_t width, const char *text) {
     size_t n = strlen(text);
     if (n > width)
-        n = width - 1;
+        n = width;
     memcpy(dst, text, n);
     size_t i = n;
-    while (i <= width)
+    while (i < width)
         dst[i++] = ' ';
     dst[i] = '\0';
 }
