diff --git a/flusher.go b/flusher.go
index e44d943..614936c 100644
--- a/flusher.go
+++ b/flusher.go
@@ -50,6 +50,8 @@ func (f *Flusher) Close() {
 
 func (f *Flusher) loop() {
 	defer close(f.done)
+	ticker := time.NewTicker(f.interval)
+	defer ticker.Stop()
 	var batch []Point
 	flush := func() {
 		if len(batch) == 0 {
@@ -65,7 +67,7 @@ func (f *Flusher) loop() {
 			if len(batch) >= f.max {
 				flush()
 			}
-		case <-time.Tick(f.interval):
+		case <-ticker.C:
 			flush()
 		case <-f.stop:
 			flush()
