diff --git a/src/controller.cpp b/src/controller.cpp
--- a/src/controller.cpp
+++ b/src/controller.cpp
@@ -84,6 +84,10 @@ init_result initialize_controller(const controller_config& config,
     void* const decoder_memory =
         allocator.allocate(sizeof(detail::decoder), alignof(detail::decoder));
     if (decoder_memory == nullptr) {
+        transport->~transport();
+        allocator.deallocate(transport_memory,
+                             sizeof(detail::transport),
+                             alignof(detail::transport));
         return {nullptr, init_error::out_of_memory};
     }
     detail::decoder* const decoder =
@@ -97,6 +101,10 @@ init_result initialize_controller(const controller_config& config,
         allocator.deallocate(decoder_memory,
                              sizeof(detail::decoder),
                              alignof(detail::decoder));
+        transport->~transport();
+        allocator.deallocate(transport_memory,
+                             sizeof(detail::transport),
+                             alignof(detail::transport));
         return {nullptr, init_error::out_of_memory};
     }
 
