diff --git a/test/TestBootstrap.ps1 b/test/TestBootstrap.ps1
index dc9fc2f..33f7884 100644
--- a/test/TestBootstrap.ps1
+++ b/test/TestBootstrap.ps1
@@ -9,5 +9,10 @@ function Import-ModuleUnderTest {
         [string] $Name
     )
 
+    $loadedModule = Get-Module -Name $Name
+    if ($null -ne $loadedModule) {
+        Remove-Module -Name $Name -Force -ErrorAction Stop
+    }
+
     Import-Module -Name $Path -Global -ErrorAction Stop
 }
