diff --git a/Directory.Build.props b/Directory.Build.props
index 02f9976..5e55665 100644
--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -18,7 +18,7 @@
 
   <!-- These settings describe artifacts published by the Release pipeline. -->
   <Choose>
-    <When Condition="'$(Configuration)' == 'Debug'">
+    <When Condition="'$(Configuration)' == 'Release'">
       <PropertyGroup>
         <PackageId>Moonshiner.Deterministic.Text</PackageId>
         <Authors>Moonshiner Maintainers</Authors>
